internal/core/src/index/TextMatchIndex.cpp

```diff
@@ -135,8 +151,10 @@ TextMatchIndex::shouldTriggerCommit() {
 
 void
 TextMatchIndex::Commit() {
-    wrapper_->commit();
-    last_commit_time_.store(stdclock::now());
+    if (mtx_.try_lock()) {
+        wrapper_->commit();
+        last_commit_time_.store(stdclock::now());
```

Comment:
need call mtx_.unlock()

Reply:
Use unique_lock is better.
[std::unique_lock](http://en.cppreference.com/w/cpp/thread/unique_lock)<[std::mutex](http://en.cppreference.com/w/cpp/thread/mutex)> lock(mtx_, [std::defer_lock](http://en.cppreference.com/w/cpp/thread/lock_tag));
if lock.try_lock() {
...
}

Reply:
It seems that we can also use [std::recursive_mutex](http://en.cppreference.com/w/cpp/thread/recursive_mutex) to ensure safety

Reply:
fixed